// @mongodb-js/saslprep/dist/memory-code-points.js
"use strict";
var __importDefault = undefined && undefined.__importDefault || function (mod) {
    return mod && mod.__esModule ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createMemoryCodePoints = void 0;
var sparse_bitfield_1 = __importDefault(require("sparse-bitfield"));
function createMemoryCodePoints(data) {
    var offset = 0;
    function read() {
        var size = data.readUInt32BE(offset);
        offset += 4;
        var codepoints = data.slice(offset, offset + size);
        offset += size;
        return (0, sparse_bitfield_1["default"])({ buffer: codepoints });
    }
    var unassigned_code_points = read();
    var commonly_mapped_to_nothing = read();
    var non_ASCII_space_characters = read();
    var prohibited_characters = read();
    var bidirectional_r_al = read();
    var bidirectional_l = read();
    return {
        unassigned_code_points: unassigned_code_points,
        commonly_mapped_to_nothing: commonly_mapped_to_nothing,
        non_ASCII_space_characters: non_ASCII_space_characters,
        prohibited_characters: prohibited_characters,
        bidirectional_r_al: bidirectional_r_al,
        bidirectional_l: bidirectional_l
    };
}
exports.createMemoryCodePoints = createMemoryCodePoints;
